service/history/decisionChecker_test.go

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package history
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/suite"
+	workflow "github.com/uber/cadence/.gen/go/shared"
+	"github.com/uber/cadence/common"
+)
+
+type (
+	decisionAttrValidatorSuite struct {
+		suite.Suite
+
+		maxIDLengthLimit int
+		validator        *decisionAttrValidator
+	}
+)
+
+func TestDecisionAttrValidatorSuite(t *testing.T) {
+	s := new(decisionAttrValidatorSuite)
+	suite.Run(t, s)
+}
+
+func (s *decisionAttrValidatorSuite) SetupSuite() {
+}
+
+func (s *decisionAttrValidatorSuite) TearDownSuite() {
+
+}
+
+func (s *decisionAttrValidatorSuite) SetupTest() {
+	s.maxIDLengthLimit = 1000
+	s.validator = newDecisionAttrValidator(s.maxIDLengthLimit)
+}
+
+func (s *decisionAttrValidatorSuite) TearDownTest() {
+
+}
+
+func (s *decisionAttrValidatorSuite) TestValidateSignalExternalWorkflowExecutionAttributes() {
```

Comment:
We should move/add tests for the other cases here too.. 

If we need to add new tests, then that can be as a separate diff..

Reply:
this is pure refactoring + moving code

Reply:
Ok. Let's add a `TODO` to add tests for the other validations - and open an issue?

Reply:
#1948